### PR TITLE
*: change the log package to logrus

### PR DESCRIPTION
**Pull request by coocood**
PD changed the log package to `logrus` a few month ago, we should use the same log package in TiDB. Otherwise, the logs written by PD client will be lost when logging to a file.

### CODE REVIEW

**Comment by coocood** [comment]
@andelf 
I want to keep `ngaut/log` package for a while in case it is used somewhere unexpectedly.

File: _vendor/src/github.com/pingcap/pd/pkg/logutil/log.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		MaxBackups: cfg.MaxBackups,
+		MaxAge:     cfg.MaxDays,
+		LocalTime:  true,
+	}
+
+	log.SetOutput(output)
+	return nil
+}
+
+// InitLogger initalizes PD's logger.
```
[the remark this answers is not in the document]
Comment (coocood) [reply _vendor/src/github.com/pingcap/pd/pkg/logutil/log.go]: But it's in vendor, we are not supposed to touch the code in vendor.

File: _vendor/src/github.com/pingcap/pd/pkg/logutil/log.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/pkg/capnslog"
+	"github.com/juju/errors"
+	lumberjack "gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	defaultLogTimeFormat = "2006/01/02 15:04:05.000"
+	defaultLogMaxSize    = 300 // MB
```
[the remark this answers is not in the document]
Comment (coocood) [reply _vendor/src/github.com/pingcap/pd/pkg/logutil/log.go]: No.

**Comment by iamxy** [comment]
CI failed without `_vendor/src/github.com/ngaut/log/`
:(

**Comment by coocood** [comment]
@shenli @andelf PTAL

File: cmd/benchfilesort/main.go [hunk 3]
```diff
@@ -392,7 +393,10 @@ func driveRunCmd() {
 }
 
 func init() {
-	log.SetLevelByString(logLevel)
+	logLevel := os.Getenv("log_level")
```
Comment (winkyao) [inline cmd/benchfilesort/main.go]: why not just use logLevel? it is defined as `logLevel    = "warn"`

**Comment by coocood** [comment]
@winkyao PTAL

File: store/tikv/backoff.go [hunk 4]
```diff
@@ -179,7 +179,7 @@ func (b *Backoffer) Backoff(typ backoffType, err error) error {
 		errMsg := fmt.Sprintf("backoffer.maxSleep %dms is exceeded, errors:", b.maxSleep)
 		for i, err := range b.errors {
 			// Print only last 3 errors for non-DEBUG log levels.
-			if log.GetLogLevel() >= log.LOG_LEVEL_DEBUG || i >= len(b.errors)-3 {
+			if log.GetLevel() == log.DebugLevel || i >= len(b.errors)-3 {
```
Comment (tiancaiamao) [inline store/tikv/backoff.go]: why this change?

File: util/systimemon/systime_mon.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+package systimemon
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// StartMonitor calls systimeErrHandler if system time jump backward.
+func StartMonitor(now func() time.Time, systimeErrHandler func()) {
```
Comment (tiancaiamao) [inline util/systimemon/systime_mon.go]: Would Go1.9 fix time jump?

File: store/tikv/backoff.go (hunk 4 above)
Comment (coocood) [reply store/tikv/backoff.go]: Use the logrus API to implement the same logic.

**Comment by tiancaiamao** [comment]
LGTM
